eng/mgmt/automation/generate.py

```diff
@@ -199,6 +207,116 @@ def sdk_automation_autorest(config: dict) -> List[dict]:
     return packages
 
 
+def sdk_automation_typespec(config: dict) -> List[dict]:
+    base_dir = os.path.abspath(os.path.dirname(sys.argv[0]))
+    sdk_root = os.path.abspath(os.path.join(base_dir, SDK_ROOT))
+    spec_root = os.path.abspath(config['specFolder'])
+
+    packages = []
+    if 'relatedTypeSpecProjectFolder' not in config:
+        return packages
+
+    head_sha: str = config['headSha']
+    repo_url: str = config['repoHttpsUrl']
+
+    tsp_projects = config['relatedTypeSpecProjectFolder']
+    if isinstance(tsp_projects, str):
+        tsp_projects = [tsp_projects]
+
+    # mgmt tsp project folder follow the pattern, e.g. specification/deviceregistry/DeviceRegistry.Management
+    arm = any(re.match(r'specification[\\/](.*)[\\/](.*)[\\.]Management', tsp_project) for tsp_project in tsp_projects)
+    if not arm:
+        return sdk_automation_typespec_data(config)
+
+    # TODO(xiaofei) extract common method from sdk_automation_typespec_data,
+    # also, support changelog
+    for tsp_project in tsp_projects:
+        tsp_dir = os.path.join(spec_root, tsp_project)
+
+        succeeded = False
+        sdk_folder = None
+        service = None
+        module = None
+        try:
+            cmd = ['pwsh', './eng/common/scripts/TypeSpec-Project-Process.ps1', tsp_dir, head_sha, repo_url]
+            logging.info('Command line: ' + ' '.join(cmd))
+            output = subprocess.check_output(cmd, cwd=sdk_root)
+            output_str = str(output, 'utf-8')
+            script_return = output_str.splitlines()[-1] # the path to sdk folder
+            sdk_folder = os.path.relpath(script_return, sdk_root)
+            logging.info('SDK folder: ' + sdk_folder)
+            if sdk_folder:
+                succeeded = True
+        except subprocess.CalledProcessError as error:
+            logging.error(f'TypeSpec-Project-Process.ps1 fail: {error}')
+
+        if succeeded:
+            # check require_sdk_integration
+            require_sdk_integration = False
+            cmd = ['git', 'add', '.']
+            check_call(cmd, sdk_root)
+            cmd = ['git', 'status', '--porcelain', os.path.join(sdk_folder, 'pom.xml')]
+            logging.info('Command line: ' + ' '.join(cmd))
+            output = subprocess.check_output(cmd, cwd=sdk_root)
+            output_str = str(output, 'utf-8')
+            git_items = output_str.splitlines()
+            if len(git_items) > 0:
+                git_pom_item = git_items[0]
+                # new pom.xml implies new SDK
+                require_sdk_integration = git_pom_item.startswith('A ')
+
+            # parse service and module
+            match = re.match(r'sdk[\\/](.*)[\\/](.*)', sdk_folder)
+            service = match.group(1)
+            module = match.group(2)
+
+            # TODO (weidxu): move to typespec-java
+            if require_sdk_integration:
+                set_or_default_version(sdk_root, GROUP_ID, module)
```

Comment:
previous failure is due to `GROUP_ID` not correct, `com.azure.resourcemanager` vs `com.azure`.